lib/helpers.js

```diff
@@ -66,26 +64,23 @@ export function setVerificationKeyPairId({verificationKeyPair, did}) {
     `${did}#${verificationKeyPair.publicKeyMultibase}`;
 }
 
-export function getKeyAgreementKeyPair({verificationKeyPair}) {
-  const contexts = [DID_CONTEXT_URL];
+export function getKeyAgreementKeyPair({contexts, verificationPublicKey}) {
   // The KAK pair will use the source key's controller, but may generate
   // its own .id
   let keyAgreementKeyPair;
 
-  switch(verificationKeyPair.type) {
+  switch(verificationPublicKey.type) {
     case 'Ed25519VerificationKey2018': {
       keyAgreementKeyPair = X25519KeyAgreementKey2019
-        .fromEd25519VerificationKey2018({keyPair: verificationKeyPair});
+        .fromEd25519VerificationKey2018({keyPair: verificationPublicKey});
       contexts.push(
```

Comment:
This can be made a single line:

```suggestion
      contexts.push(X25519KeyAgreementKey2019.SUITE_CONTEXT);
```